File: backend/ErpPortfolio.Api/Models/Fornecedor.cs
// =====================================================================================
// Arquivo....: Fornecedor.cs
// Versão.....: 1.0.0
// Data.......: 22/09/2026
// Descrição..: Entidade de domínio que representa um fornecedor (pessoa física ou jurídica).
//              Mesmo formato do Cliente (SPEC.md, F1); documento único num índice próprio,
//              independente do de clientes.
// -------------------------------------------------------------------------------------
// Banco......: PostgreSQL - erp_portfolio_db
// Tabelas....: public.fornecedores
// Fontes.....: Mapeada em ErpPortfolioDbContext.Fornecedores (EF Core / Npgsql).
//              Colunas: id, nome, documento, email, telefone, cidade, uf, ativo,
//              data_cadastro (o mapeamento de nomes fica no DbContext).
// -------------------------------------------------------------------------------------
// Histórico de alterações:
//   1.0.0 - 22/09/2026 - Criação do arquivo.
// =====================================================================================

namespace ErpPortfolio.Api.Models;

public class Fornecedor
{
    public int Id { get; set; }

    public string Nome { get; set; } = string.Empty;

    /// <summary>CPF (11 dígitos) ou CNPJ (14 caracteres), armazenado sem máscara.</summary>
    public string Documento { get; set; } = string.Empty;

    public string? Email { get; set; }

    public string? Telefone { get; set; }

    public string Cidade { get; set; } = string.Empty;

    /// <summary>Sigla da unidade federativa, sempre em maiúsculas (ex.: SP).</summary>
    public string Uf { get; set; } = string.Empty;

    public bool Ativo { get; set; } = true;

    /// <summary>Data/hora de cadastro em UTC.</summary>
    public DateTime DataCadastro { get; set; }
}

File: frontend/erp-portfolio-web/src/types/fornecedor.ts
/**
 * =====================================================================
 * Arquivo....: fornecedor.ts
 * Versão.....: 1.0.0
 * Data.......: 23/09/2026
 * Descrição..: Tipos do módulo de Fornecedores, espelhando os DTOs da API
 *              (FornecedorRespostaDto, FornecedorCriacaoDto,
 *              FornecedorAtualizacaoDto, FornecedorFiltroDto e
 *              ResultadoPaginadoDto). Espelho de types/cliente.ts.
 * ---------------------------------------------------------------------
 * Histórico de alterações:
 *   1.0.0 - 23/09/2026 - Criação do arquivo.
 * =====================================================================
 */

export interface Fornecedor {
  id: number
  nome: string
  documento: string
  email: string | null
  telefone: string | null
  cidade: string
  uf: string
  ativo: boolean
  /** Data/hora ISO 8601 em UTC. */
  dataCadastro: string
}

export interface FornecedorCriacao {
  nome: string
  documento: string
  email: string | null
  telefone: string | null
  cidade: string
  uf: string
}

export interface FornecedorAtualizacao extends FornecedorCriacao {
  ativo: boolean
}

export interface FornecedorFiltro {
  nome?: string
  ufs?: string[]
  /** true = só ativos, false = só inativos, ausente = todos. */
  ativo?: boolean
  pagina: number
  tamanhoPagina: number
}

export type { ResultadoPaginado } from './paginacao'
